### ISSUE TITLE
SC 3.3.8 Redundant Entry and User Frustration

### ISSUE THREAD

**Issue by WegnerTR** [w3c/wcag#1431]
On behalf of Thomson Reuters:

Thomson Reuters recognizes the value of eliminating redundant input. We feel that this value could be preserved and the user experience improved by adding an additional exception where redundant entry is acceptable if the redundant input is designed to is designed to prevent user frustration from committing a critical form input error that will make it difficult and time consuming for the user to recover. For example, if a form takes an email address and acts upon that information with no further confirmation steps, asking the user to input their email twice serves as a mechanism to prevent the user from having the form acted upon in error.

**Comment by WilcoFiers** [comment]
**Proposed response**: Re-entry of an e-mail address for the purpose of preventing error is considered to be part of the exception of 3.3.8 Redundant entry. To clarify, we can make the following change:

> - Exception: When re-entering the information is essential.
> + Exception: When re-entering the information is essential, or unless re-entry is used to prevent user error, such as re-entry of a new password.

This solution would also resolve my own comment: #1346

**Comment by Helixopp** [comment]
+1 to Wilco's reply. We would consider confirming contact information, email, login credentials, essential/critical info, etc. as essential. Therefore, it would be an exception. the purpose of this SC is primarily, if not wholly, concerned with redundant requests for information requests that occur in future steps, after the user has already progressed further in the process, and no longer has that previous information before them to reference.

**Comment by alastc** [comment]
Overall if something is requested twice in the same step, it is not covered by the SC text (because it only applies to subsequent steps).

It we added an exception for password, then I'd suggest:
<hr>
Exception: When re-entering the information is essential, or when the information is not available for security reasons such as re-entering a new password.
<hr>

However, I do think that might raise as many questions as it answers, because it is redundant if you think about steps in the way intended.

**Comment by Helixopp** [comment]
Maybe this is better addressed in the understanding document. I think explaining ourselves there is the best option

This message was Sent from my iPhone. Please excuse any typographic errors.

On Dec 1, 2020, at 10:15 AM, Alastair Campbell <notifications@github.com> wrote:

﻿

Overall if something is requested twice in the same step, it is not covered by the SC text (because it only applies to subsequent steps).

It we added an exception for password, then I'd suggest:

________________________________
Exception: When re-entering the information is essential, or when the information is not available for security reasons such as re-entering a new password.
________________________________

However, I do think that might raise as many questions as it answers, because it is redundant if you think about steps in the way intended.

**Comment by alastc** [comment]
Noting a comment from @mbgower in the survey:
<hr>
I'd argue re-entering a new password _is_ essential and is covered by the existing shorter exception. As well, a double-email in some contexts can clearly be argued as essential.
I think this can be addressed in the Understanding document, and there is no need to update the normative text or note.
I'm a bit leery of language that suggests a 'step could be a page' as that could lead to some very unfortunate interpretations.

**Comment by alastc** [comment]
New draft response as we have updated the SC text to remove 'step':
<hr>
The group agreed to leave the SC text as it is: https://w3c.github.io/wcag/understanding/redundant-entry.html

Key factors were that: 
- We didn't want to bake-in a particular solution that might change in future. I.e. in future it may not be essential to have a 2nd password, e.g. if you rely on biometrics, or can require the use of a password manager.
- There is pretty universal agreement that having a second password entry is essential for it's purpose. 
-  Given that, it appears to meet the definition: "if removed, would fundamentally change the information or functionality of the content, and information and functionality cannot be achieved in another way that would conform"

As mitigations against confusion we already have:

- A note underneath the SC text: "Security verification, such as repeating a password, is considered essential."
- A line in the intent: "There is an exception for essential uses of input re-entry for things like password entry (security), or memory games which would not be possible if the previous answers were supplied."
